recipes/andreasbuhr-cppcoro/all/conanfile.py

```diff
@@ -41,72 +35,75 @@ def _minimum_compilers_version(self):
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
-    def _configure_cmake(self):
-        if not self._cmake:
-            self._cmake = CMake(self)
-            if self.settings.os == "Windows" and self.options.shared:
-                self._cmake.definitions["CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS"] = "ON"
-            self._cmake.configure()
-        return self._cmake
+    def generate(self):
+        tc = CMakeToolchain(self)
+        if self.settings.os == "Windows" and self.options.shared:
+            tc.variables["CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS"] = "ON"
+        tc.generate()
 
     def build(self):
-        cmake = self._configure_cmake()
+        cmake = CMake(self)
+        cmake.configure()
         cmake.build()
 
     def package(self):
-        self.copy("LICENSE.txt", dst="licenses", src=self._source_subfolder)
-        cmake = self._configure_cmake()
+        copy(self, "LICENSE.txt", dst="licenses", src=self.source_folder)
+        cmake = CMake(self)
         cmake.install()
-        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
 
     def package_info(self):
-        self.cpp_info.filenames["cmake_find_package"] = "cppcoro"
-        self.cpp_info.filenames["cmake_find_package_multi"] = "cppcoro"
-        self.cpp_info.names["cmake_find_package"] = "cppcoro"
-        self.cpp_info.names["cmake_find_package_multi"] = "cppcoro"
-
-        comp = self.cpp_info.components["cppcoro"]
-        comp.names["cmake_find_package"] = "cppcoro"
-        comp.names["cmake_find_package_multi"] = "cppcoro"
-        comp.libs = ["cppcoro"]
+        self.cpp_info.libs = ["cppcoro"]
 
         if self.settings.os == "Linux" and self.options.shared:
-            comp.system_libs = ["pthread"]
+            self.cpp_info.system_libs = ["pthread"]
         if self.settings.os == "Windows":
-            comp.system_libs = ["synchronization"]
+            self.cpp_info.system_libs = ["synchronization"]
 
-        if self.settings.compiler == "Visual Studio":
-            comp.cxxflags.append("/await")
+        if self.settings.compiler == "msvc":
+            self.cpp_info.cxxflags.append("/await")
         elif self.settings.compiler == "gcc":
-            comp.cxxflags.append("-fcoroutines")
-            comp.defines.append("CPPCORO_COMPILER_SUPPORTS_SYMMETRIC_TRANSFER=1")
-        elif self.settings.compiler == "clang" or self.settings.compiler == "apple-clang":
-            comp.cxxflags.append("-fcoroutines-ts")
+            self.cpp_info.cxxflags.append("-fcoroutines")
+            self.cpp_info.defines.append("CPPCORO_COMPILER_SUPPORTS_SYMMETRIC_TRANSFER=1")
+        elif self.settings.compiler in ["clang", "apple-clang"]:
+            self.cpp_info.cxxflags.append("-fcoroutines-ts")
```

Comment:
This flag is no longer valid in newer clang versions afaik, should check its version beforhand

Reply:
Per comment above, apple-clang 14 (and 15, after upgrading) works as-is, so I'm going to leave it.

Reply:
> (no compiler explorer :( )

Damn, let me try to see if there's something I can do on that front, I'm also a main collaborator there!